src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ConstructedEETypeNode.cs

```diff
@@ -168,6 +166,8 @@ protected override DependencyList ComputeNonRelocationBasedDependencies(NodeFact
             // Ask the metadata manager if we have any dependencies due to reflectability.
             factory.MetadataManager.GetDependenciesDueToReflectability(ref dependencyList, factory, _type);
 
+            CodeBasedDependencyAlgorithm.AddInterestingInteropConstructedTypeDependencies(ref dependencyList, factory, _type);
```

Comment:
> We don't only add dependencies for the delegates on this code path . So we should be fine for now.

Should it call a different method then? Once you uncomment the commented out code in the method it calls, we'll be generating useless interop code for all boxed structs, no? We don't have CI tooling to detect unintentional size on disk regressions, so the only way around that is to structure code in a way where they don't happen.